Sprint 22

KiMate34 opened the pull request:
Sprint 22

KiMate34 committed:
Create fragment_creation_window_playlist.xml
Add new String in strings.xml
Create transition on fragment_creation_window_playlist.xml in main_navigation_graph.xml
Add playlist_image.xml in drawable

KiMate34 committed:
Create PlaylistDao, PlaylistEntity
Add entity in AppDatabase.kt

KiMate34 committed:
Add playlistDao in AppDatabase.kt
Create repository PlaylistRepository and PlaylistRepositoryImpl

KiMate34 committed:
Add package model from library/domain
Add data class Playlist.kt
Create PlaylistInteractor.kt and PlaylistInteractorImpl.kt

KiMate34 committed:
Create class CreationPlaylist and CreationPlaylistViewModel
Change field uri for Playlist.kt and PlaylistEntity.kt
Add method savePlaylist in interactor and repository
Create PlaylistDatabaseConverter.kt

KiMate34 committed:
Add new graph in main_navigation_graph.xml
Add listener from button new playlist
Change tools context in fragment_creation_window_playlist.xml

KiMate34 committed:
Add data layer, domain layer and ui layer from FragmentCreationPlaylist

KiMate34 committed:
Add data layer, domain layer and ui layer from FragmentFavoriteTracks

KiMate34 committed:
UI layer from FragmentFavoriteTracks

KiMate34 committed:
Delete AudioPlayerActivity, create FragmentAudioPlayer
Add data layer, domain layer, ui layer FragmentAudioPlayer

KiMate34 committed:
Add new dependency in DI

KiMate34 committed:
Add new style, and new strings from new fragments

KiMate34 committed:
Change navigation from SearchFragment
Add condition from navigation in MainActivity

KiMate34 committed:
Delete CreationPlaylist.kt, AudioPlayerActivity.kt, activity_audioplayer.xml
Change PlaylistDao, FragmentAudioPlayer, TrackViewModel
Add new style in styles.xml
Add new string in strings.xml

kosstyango commented:
⚠️ Критические замечания:
1. Кнопка "Создать" в активном состоянии - такая же серая, как и в НЕактивном состоянии. Должна быть синяя.
2. Активная рамка на экране Новый плейлист должна быть СИНЯЯ, а не фиолетовая (см.скрин).
![Спринт_22_Матвеев3](https://github.com/user-attachments/assets/aea11103-3178-4c2c-96b6-165eec563146)

3. При выходе из создания плейлиста из Аудиоплеера по кнопке "Назад" (и по системной кнопке BACK) приложение "падает" (скрин ошибок)
![Спринт_22_Матвеев4](https://github.com/user-attachments/assets/c94246f1-b2dc-4f0a-977f-1b651ef81b77)

4. При нажатии кнопки "Назад" (в левом верхнем углу) на экране Новый плейлист сначала появляется системный диалог, после которого приложение "падает". При нажатии на системную кнопку BACK системный диалог НЕ появляется, приложение "падает" сразу.
5. Треки с ОЧЕНЬ длинным названием отображаются неправильно: строка с названием переносится, а не обрезается (см.скрин)
![Спринт_22_Матвеев2](https://github.com/user-attachments/assets/ae168490-fdf7-4379-9006-51511a00e2e6)

6. На экране Новый плейлист в темной теме не отображаются подсказки в неактивных полях Название и Описание (см.скрин)
![Спринт_22_Матвеев1](https://github.com/user-attachments/assets/7345b4d0-3b86-4510-817a-87806835cfc2)


🍏 Рекомендации, не влияющие на результат проверки работы:
1. Всплывающее окно добавления трека в плейлист всплывает невысоко: всего 1 плейлист в нем видно. На макете в Фигма четко нарисовано: всплывать окно должно на 2/3 экрана.
2. Кнопка "Создать" на экране Новый плейлист выполнена CAPSLOCK'ом - это не соответствует дизайну.
3. Шрифт на кнопке "Очистить историю" очень мелкий - не соответствует дизайн-макету в Фигма.

KiMate34 committed:
Change track name in activity_track.xml
Add new xml class colors.xml
Add new drawable create_button.xml
Add new color create_text_view_color.xml
Change edit text in fragment_creation_window_playlist.xml

KillReal34 commented:
Со всем разобрался. Единственное не понял почему у тебя приложение падает((( Я не понимаю, при каком сценарии это происходит, ибо делаю я всё тоже самое. Тобиж, при выходе из фрагмента создания плейлиста - краш не происходит(

KiMate34 committed:
Fix bug navigation from FragmentCreationPlaylist to FragmentAudioPlayer

kosstyango commented:
Ты отлично поработал, исправил почти все критические замечания и даже один из НЕкритических, молодец!!
⚠️ Критические замечания:
1. Кнопка "Создать" в активном состоянии - такая же серая, как и в НЕактивном состоянии. Должна быть синяя.
2. Активная рамка на экране Новый плейлист должна быть СИНЯЯ, а не фиолетовая (см.скрин в PR).
3. При выходе из создания плейлиста из Аудиоплеера по кнопке "Назад" (и по системной кнопке BACK) приложение "падает" (скрин ошибок приложу в PR)
4. При нажатии кнопки "Назад" (в левом верхнем углу) на экране Новый плейлист сначала появляется системный диалог, после которого приложение "падает". Но при нажатии на системную кнопку BACK системный диалог НЕ появляется, приложение "падает" сразу. Теперь приложение не "падает", но диалога по-прежнему нет при выходе из создания плейлиста по кнопке BACK
5. Треки с ОЧЕНЬ длинным названием отображаются неправильно: строка с названием переносится, а не обрезается (см.скрин в PR)
6. На экране Новый плейлист в темной теме не отображаются подсказки в неактивных полях Название и Описание (см.скрин в PR)
🍏 Рекомендации, не влияющие на результат проверки работы:
1. Всплывающее окно добавления трека в плейлист всплывает невысоко: всего 1 плейлист в нем видно. На макете в Фигма четко нарисовано: всплывать окно должно на 2/3 экрана. Это осталось. чтобы увидеть второй и следующие плейлисты приходится "поднимать" окно до верха.
2. Кнопка "Создать" на экране Новый плейлист выполнена CAPSLOCK'ом - это не соответствует дизайну.
3. Шрифт на кнопке "Очистить историю" очень мелкий - не соответствует дизайн-макету в Фигма.Это пока осталось
Однако, в процессе исправлений ВОЗНИК НОВЫЙ ДЕФЕКТ: кнопка PLAY на экране Плейлист теперь НЕ работает - ни на физическом, ни на виртуальном устройстве :(

KiMate34 committed:
Fix bug for button Play on TrackViewModel.kt

KiMate34 committed:
Add condition from system button Back for FragmentCreationPlaylist.kt

KiMate34 committed:
Change text size for btn_clear_history

KiMate34 committed:
Change bottom sheet expended ratio

kosstyango commented:
Ты отлично поработал, исправил последний "крит" и даже одно из НЕкритических замечаний, молодец!!
⚠️ Критические замечания:
~~НОВЫЙ ДЕФЕКТ: кнопка PLAY на экране Плейлист теперь НЕ работает - ни на физическом, ни на виртуальном устройстве :(~~
🍏 Рекомендации, не влияющие на результат проверки работы:
~~1. Всплывающее окно добавления трека в плейлист всплывает невысоко: всего 1 плейлист в нем видно. На макете в Фигма четко нарисовано: всплывать окно должно на 2/3 экрана - чтобы увидеть второй и следующие плейлисты приходится "поднимать" окно до верха.~~
2. Шрифт на кнопке "Очистить историю" очень мелкий - не соответствует дизайн-макету в Фигма. **Теперь текст выполнен CAPS LOCK'ом и он расположен в две строки - это не соответствует дизайну в Фигма (см.скрин)**
![Спринт_22_Павлов1](https://github.com/user-attachments/assets/484dde92-265b-47c4-ac38-dc348e5d65d1)
